docs/adr/006_cups.md

```diff
@@ -0,0 +1,297 @@
+
+More information on the protocol can be found [here](https://doc.sm.tc/station/cupsproto.html).
+
+## In-scope
+
+This document focuses on:
+
+- Defining a flow diagram for common credential management scenarios:
+  - First connection
+  - Certificate rotation process
+- Defining the changes needed in IoT Hub for handling Basics Station specific information
+- Defining the changes needed in 'Facade' Azure Function for handling updated credential retrieval
+- Defining the changes needed in LoRaWan Network Server for handling above mentioned flows
+- Defining the changes needed in LoRa Device Provisioning CLI for handling creation/update of concentrators with certificate support
+
+## Out-of-scope
+
+Out of scope for this document is:
+
+- Defining a flow diagram for firmware update process
+- Defining any change needed in IoT Hub / Azure Function / Network Server for handling firmware updates
+
+## Authentication mechanisms with LoRa Basics™ Station
+
+### Context
+
+To better understand how the CUPS Protocol should be implemented, it is needed to understand how Basics Station is handling authentication.
+
+LoRa Basics™ Station supports four different authentication modes:
+
+- No Authentication
+- TLS Server Authentication
+- TLS Server and Client Authentication
+- TLS Server Authentication and Client Token
+
+An authentication mode can be set by configuring some files in the device where Basics Station is going to be executed.
+
+Basics Station can either use two different sets of credentials for achieving client authentication to the CUPS endpoint and to the LNS endpoint or re-use the same set of credentials for both.
+
+The aim of this document section is to take and record a decision for:
+
+- Which authentication mode needs to be used for connecting to CUPS endpoint
+- Which authentication mode needs to be used for connecting to LNS endpoint
+- Where are the certificates stored and how to retrieve those?
+
+### Decision(s)
+
+Out of the four authentication modes we should aim for "TLS Server and Client Authentication" for both CUPS and LNS endpoints.
+
+Driving this decision is the willing of increasing the overall security of the system.
+
+In order to keep the starter kit simple, the same set of server and client credentials are to be used for both CUPS and LNS protocol endpoints.
+
+### Authentication to CUPS endpoint
+
+The CUPS endpoint should support mutual TLS authentication between client (basic station) and server (network server IoT Edge module).
+
+If not manually disabled via flags, for increased security, the server has to "require" client authentication and verify the provided credentials.
+
+Therefore, in order for Basic Station to connect to the CUPS HTTPS endpoint the following files should be provided to the station:
+
+- cups.uri
+- cups.crt (including the certificate in PEM format)
```

Comment:
Why not use DER format directly as described in the spec?

Reply:
Where in the spec is it saying it's DER?

Reply:
https://lora-developers.semtech.com/build/software/lora-basics/lora-basics-for-gateways/?url=cupsproto.html

> For client authentication using X509 certificates, all credentials are encoded in DER format:
trust is the certificate of the trusted certificate authority (CA)
cert is the personal certificate for the gateway
key is the personal private key for the gateway

Reply:
https://lora-developers.semtech.com/build/software/lora-basics/lora-basics-for-gateways/?url=credentials.html

Here we have a clear indication of the fact that the *.trust is in PEM-encoded format.

Also having a look at the script in the [examples from BasicStation repo](https://github.com/lorabasics/basicstation/blob/master/examples/cups/prep.sh) we can see that only the signing keys for firmware updates are output as DER.

As agreed, let's keep this PEM 👍